File: app/src/main/java/ir/androidpower/zseekbar/sample/MainActivity.java
package ir.androidpower.zseekbar.sample;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import zHelper.view.ZSeekBar;


public class MainActivity extends AppCompatActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);


    final TextView test_progress = findViewById(R.id.test_progress2);
    final ZSeekBar test_zSeekBar = findViewById(R.id.test_zSeekBar2);


    test_zSeekBar.setOnZArcSeekBarChangeListener(new ZSeekBar.OnZArcSeekBarChangeListener() {
      @Override
      public void onProgressChanged(ZSeekBar circleProgress, float progress, boolean fromUser) {
        test_progress.setText((int) progress + "");
      }


      @Override
      public void onStartTrackingTouch(ZSeekBar circleProgress) {

      }


      @Override
      public void onStopTrackingTouch(ZSeekBar circleProgress) {

      }
    });
  }
}

File: app/src/main/java/zabih/lib/zseekbar/sample/MainActivity.kt
package zabih.lib.zseekbar.sample

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import zHelper.view.ZSeekBar
import zHelper.view.ZSeekBar.OnZArcSeekBarChangeListener


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val testProgress = findViewById<TextView>(R.id.test_progress2)
        val zSeekBar = findViewById<ZSeekBar>(R.id.test_zSeekBar2)

        zSeekBar.setOnZArcSeekBarChangeListener(object : OnZArcSeekBarChangeListener {
            override fun onProgressChanged(
                circleProgress: ZSeekBar,
                progress: Float,
                fromUser: Boolean
            ) {
                testProgress.text = progress.toInt().toString()
            }


            override fun onStartTrackingTouch(circleProgress: ZSeekBar) {
            }


            override fun onStopTrackingTouch(circleProgress: ZSeekBar) {
            }
        })
    }
}